cloud-sql/postgres/Program.cs

```diff
@@ -14,21 +14,67 @@
  * the License.
  */
 
+using CloudSql.Settings;
+using Google.Cloud.Diagnostics.AspNetCore;
 using Microsoft.AspNetCore;
 using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.IO;
 
 namespace CloudSql
 {
     public class Program
     {
+        public static AppSettings AppSettings { get; private set; }
+
         public static void Main(string[] args)
         {
-            BuildWebHost(args).Run();
+            BuildWebHost(args).Build().Run();
         }
 
-        public static IWebHost BuildWebHost(string[] args) =>
-            WebHost.CreateDefaultBuilder(args)
+        public static IWebHostBuilder BuildWebHost(string[] args)
+        {
+            ReadAppSettings();
+
+            return WebHost.CreateDefaultBuilder(args)
+                .UseGoogleDiagnostics(AppSettings.GoogleCloudSettings.ProjectId,
+                        AppSettings.GoogleCloudSettings.ServiceName,
+                        AppSettings.GoogleCloudSettings.Version)
                 .UseStartup<Startup>()
+                .UsePortEnvironmentVariable();
+        }
+
+        /// <summary>
+        /// Read application settings from appsettings.json. 
+        /// </summary>
+        private static void ReadAppSettings()
+        {
+            var config = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("appsettings.json", optional: false)
                 .Build();
+
+            // Read json config into AppSettings.
+            AppSettings = new AppSettings();
+            config.Bind(AppSettings);
+        }
+    }
+
+    static class ProgramExtensions
+    {
+        // Google Cloud Run sets the PORT environment variable to tell this
+        // process which port to listen to.
+        public static IWebHostBuilder UsePortEnvironmentVariable(
+            this IWebHostBuilder builder)
+        {
+            string port = Environment.GetEnvironmentVariable("PORT");
+            if (!string.IsNullOrEmpty(port))

```

Comment:
Same comment here as in the other one, not sure if there should be an 8080 fallback value for the port.

Reply:
I attempted to update the command to be
string port = Environment.GetEnvironmentVariable("PORT") ?? "8080";

but encountered the same issue as noted in my comment on the Cloud SQL Postgres PR.
This suggested change causes the app to fail to run locally. 

As it currently is the app successfully works as expected running locally, on Cloud Run
and on App Engine.

Please force merge if everything else looks okay. Thanks!

Reply:
OK, I've let the tests run now, they should be green, and you can merge on green. Thanks!!